src/server.cpp

```diff
@@ -2516,10 +2516,14 @@ bool Server::addMediaFile(const std::string &filename,
 
 	// Read data
 	std::string filedata;
-	if (!fs::ReadFile(filepath, filedata)) {
-		errorstream << "Server::addMediaFile(): Failed to open \""
-					<< filename << "\" for reading" << std::endl;
-		return false;
+	if (filedata_to && !(*filedata_to).empty()) {
+		filedata = *filedata_to;
```

Comment:
I believe this creates a copy of filedata_to, rather than setting filedata to point to filedata_to.
This is a waste, if someone more experienced with C++ could point the correct method out that'd be nice :)

hmm… maybe I can avoid the two variables altogether?

Reply:
You could add a string to the stack where you can store the filedata (if you read a file), and then take a reference to the string that you use:
```cpp
std::string filedata_file;
std::string &filedata = [&] {
if (filedata_to && !filedata_to->empty()) {
return *filedata_to;
} else {
 fs::ReadFile(filepath, filedata_file);
return filedata_file;
}
}();
```

Reply:
Had to change to `const std::string &filedata`, but appears to work.
Can't say I fully understand - isn't `*filedata_to` returning by value and therefore making a copy?
…maybe I should spend some time actually learning C++ :P

Reply:
Oh, oops, you're right.
There needs to be an explicit return type to return references.
I've updated my comment.